src/components/PublicMentorPage/PublicMentorPage.tsx

```diff
@@ -0,0 +1,171 @@
+import { type MentorCardType } from '../../types';
+import { API_URL } from '../../constants';
+
+const PublicMentorPage: FC = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const queryParams = new URLSearchParams(location.search);
+  const categoryParam = queryParams.get('category');
+
+  const [mentors, setMentors] = useState<MentorCardType[]>([]);
+  const [categories, setCategories] = useState<string[]>([]);
+  const [selectedCategory, setSelectedCategory] = useState<string | null>(
+    categoryParam
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    const mentorsUrl = `${API_URL}/mentors`;
+    const categoriesUrl = `${API_URL}/categories`;
+
+    Promise.all([
+      fetch(mentorsUrl).then(async (response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return await response.json();
+      }),
+      fetch(categoriesUrl).then(async (response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return await response.json();
+      }),
+    ])
+      .then(([mentorsData, categoriesData]) => {
+        setMentors(mentorsData.mentors);
+
+        const categoryNames: string[] = categoriesData.categories.map(
+          (category: { category: string }) => category.category
+        );
+        const uniqueCategories: string[] = Array.from(new Set(categoryNames));
+        setCategories(uniqueCategories);
+
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.error('Error fetching data:', error);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const handleSortAZ = (): void => {
+    setIsLoading(true);
+    const sortedMentors = [...mentors].sort((a, b) =>
+      a.profile.first_name.localeCompare(b.profile.first_name)
+    );
+    setMentors(sortedMentors);
+    setIsLoading(false);
+  };
+
+  const handleCategoryChange = (category: string | null): void => {
+    setSelectedCategory(category);
+    setIsLoading(true);
+    const apiUrl = `${API_URL}/mentors${
+      category !== null ? `?category=${category}` : ''
+    }`;
+    fetch(apiUrl)
+      .then(async (response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return await response.json();
+      })
+      .then((data) => {
+        setMentors(data.mentors);
+        setError(null);
+        setIsLoading(false);
+        navigate({ search: category !== null ? `?category=${category}` : '' });
+      })
+      .catch((error) => {
+        if (error.message === 'Mentors not found') {
+          setError('No mentors found for this category.');
+        } else {
+          setError('No mentors found for this category.');
+        }
+        setIsLoading(false);
+      });
+  };
```

Comment:
Take a look at this https://tanstack.com/query/latest/docs/framework/react/guides/queries. Also we are using axios instead of fetch

```
const { data, error, isLoading } = useQuery(
  ['mentors', selectedCategory],
  async () => {
    if (!selectedCategory) return;
    const apiUrl = `${API_URL}/mentors${selectedCategory ? `?category=${selectedCategory}` : ''}`;
    const response = await axios.get(apiUrl);
    if (!response.data) {
      throw new Error('Network response was not ok');
    }
    const data = response.data;
    if (data.mentors.length === 0) {
      throw new Error('Mentors not found');
    }
    return data.mentors;
  }
);
```